### ISSUE TITLE
Test failure: IndexError in wiener_filter

### ISSUE THREAD

**Issue by olebole** [astroML/astroML#101]
Hi,

since some time, two tests fail in the Debian package; both when building the package and when running the tests on the installed package:
```
======================================================================
ERROR: astroML.tests.test_filters.test_wiener_filter_simple
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/1st/astroml-0.3/.pybuild/pythonX.Y_2.7/build/astroML/tests/test_filters.py", line 27, in test_wiener_filter_simple
    s = wiener_filter(t, h)
  File "/build/1st/astroml-0.3/.pybuild/pythonX.Y_2.7/build/astroML/filters.py", line 186, in wiener_filter
    width_guess = np.min(np.abs(f[PSD[1:] < np.mean(PSD[1:])]))
IndexError: boolean index did not match indexed array along dimension 0; dimension is 256 but corresponding boolean dimension is 255

======================================================================
ERROR: astroML.tests.test_filters.test_wienter_filter_spike
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/build/1st/astroml-0.3/.pybuild/pythonX.Y_2.7/build/astroML/tests/test_filters.py", line 39, in test_wienter_filter_spike
    h_smooth = wiener_filter(t, hN)
  File "/build/1st/astroml-0.3/.pybuild/pythonX.Y_2.7/build/astroML/filters.py", line 186, in wiener_filter
    width_guess = np.min(np.abs(f[PSD[1:] < np.mean(PSD[1:])]))
IndexError: boolean index did not match indexed array along dimension 0; dimension is 2048 but corresponding boolean dimension is 2047

----------------------------------------------------------------------
Ran 214 tests in 5.467s

FAILED (errors=2)
```

This happens with Python 2.7. Scipy version is 0.18.1 and 0.19; scikit-learn version 0.19.
The patches d8e6117cbfaeb907680de153daa5ebd0b5338f8c and 7d03226f48dc772b3a751e203123d5418187aaeb are both applied.

This is reported ad Debian bug [#877161](https://bugs.debian.org/877161).

**Comment by VikrantKamble** [comment]
Making a small change in the way width_guess is calculated in the filters.py file seems to work for me.

    # fit signal/noise params if necessary
    if signal_params is None:
        amp_guess = np.max(PSD[1:])
        width_guess = np.min(np.abs(f[1:][PSD[1:] < np.mean(PSD[1:])]))
        signal_params = (amp_guess, width_guess)
    if noise_params is None:
        noise_params = (np.mean(PSD[1:]),)

as compared to:
        `width_guess = np.min(np.abs(f[PSD[1:] < np.mean(PSD[1:])]))`

**Comment by bsipocz** [comment]
This has been fixed by #128
